app/uk/gov/hmrc/taxenrolmentassignmentfrontend/reporting/AuditEvent.scala

```diff
@@ -59,36 +47,93 @@ object AuditEvent {
       auditType = "ReportUnrecognisedAccount",
       transactionName = "reporting-unrecognised-pt-account",
       detail = getDetailsForReportingFraud(
-        request.userDetails,
-        suspiciousAccountDetails,
-        AccountTypes.PT_ASSIGNED_TO_OTHER_USER
+        suspiciousAccountDetails
       )
     )
   }
 
+  def auditSuccessfullyAutoEnrolledPersonalTax(accountType: AccountTypes.Value)
+                                              (implicit request: RequestWithUserDetailsFromSession[_]): AuditEvent = {
+
+    val optSACredentialId: Option[String] = if(request.userDetails.hasSAEnrolment || accountType == SA_ASSIGNED_TO_CURRENT_USER) {
+      Some(request.userDetails.credId)
+    } else {
+      None
```

Comment:
why arent we doing this same if condition on L70 and why does this if statement return None? surely we still want to know the other users sa cred?

Reply:
This audit event is specific to the auto/silent enrolment which is single account holders, multiple accounts with no SA enrolments on any account, and multiple accounts signed in with SA credential. The L70 function is for multiple accounts with SA on another account (as this user has been displayed pages and then made the choose to keep them separate). The if statement can return None as not all single account holders have SA or there are people with multiple accounts with no SA. We don't check for SA on another account as that is done in the L70 function.

Reply:
I've renamed both functions -  one to be auditSuccessfullyEnrolledPTWhenSAOnOtherAccount and the other auditSuccessfullyEnrolledPTWhenSANotOnOtherAccount.